app/controllers/actions/AuthAction.scala

```diff
@@ -16,28 +16,51 @@
 
 package controllers.actions
 
-import scala.concurrent.{ExecutionContext, Future}
-
 import com.google.inject.{ImplementedBy, Inject, ProvidedBy}
 import com.kenshoo.play.metrics.Metrics
+import config.AppConfig
 import controllers.routes
-import javax.inject.Provider
+import models.{IdentityData, SignedInUser}
 import models.AuthKey.{enrolment, identifierKey}
 import models.requests.AuthenticatedRequest
-import models.{IdentityData, SignedInUser}
-import play.api.mvc._
 import play.api.{Configuration, Logging}
-import uk.gov.hmrc.auth.core.retrieve.v2.Retrievals.{agentCode, _}
-import uk.gov.hmrc.auth.core.retrieve.~
+import play.api.mvc._
 import uk.gov.hmrc.auth.core.{NoActiveSession, _}
+import uk.gov.hmrc.auth.core.retrieve.v2.Retrievals.{
```

Comment:
Can these be imported at once by the _ or is it neater to use Retrievals Mixin for access?

Reply:
Indeed `import uk.gov.hmrc.auth.core.retrieve.v2.Retrievals._` seems to be possible. Any reason for this?